retrofit/src/main/java/retrofit2/ParameterHandler.java

```diff
@@ -84,17 +97,22 @@
     private final String name;
     private final Converter<T, String> valueConverter;
     private final boolean encoded;
+    private final Method method;
+    private final int p;
+
 
-    Path(String name, Converter<T, String> valueConverter, boolean encoded) {
+    Path(Method method, int p, String name, Converter<T, String> valueConverter, boolean encoded) {
       this.name = checkNotNull(name, "name == null");
```

Comment:
Would it be better if these `checkNotNull` were replaced with the `Utils.parameterError` as well?

Reply:
If they're impossible we actually usually prefer `AssertionError`. I would leave them as-is because they should never be observable by users.